python/huggingfaceserver/huggingfaceserver/completions_utils.py

```diff
@@ -0,0 +1,158 @@
+# Copyright 2024 The KServe Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import asyncio
+from http import HTTPStatus
+from typing import Dict, List, Optional, Union
+from vllm.engine.async_llm_engine import AsyncLLMEngine
+from kserve.protocol.rest.openai.types.openapi import (
+    CreateCompletionRequest,
+    ErrorResponse,
+    Error,
+    Logprobs,
+)
+from vllm.transformers_utils.tokenizer import get_tokenizer
+from vllm.sequence import Logprob
+
+
+class OpenAIServing:
+
```

Comment:
This class seems like containing vllm specific utils, it is not clear to me there is an inheritance relationship here.

Reply:
I added it as a base class initially. But it doesn't provide much value now*. I'll move it into a single class under `vllm_completions.py`